geoips/interfaces/module_based/output_checkers.py

```diff
@@ -0,0 +1,557 @@
+        """Compare the "correct" imagery found the list of current output_products.
+
+        Compares files produced in the current processing run with the list of
+        "correct" files contained in "compare_path".
+
+        Parameters
+        ----------
+        compare_path : str
+            Path to directory of "correct" products - filenames must match
+            output_products
+        output_products : list of str
+            List of strings of current output products,
+            to compare with products in compare_path
+
+        Returns
+        -------
+        int
+            Binary code: 0 if all comparisons were completed successfully.
+        """
+        global rezip
+
+        badcomps = []
+        goodcomps = []
+        missingcomps = []
+        missingproducts = []
+        compare_strings = []
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info("*** RUNNING COMPARISONS OF KNOWN OUTPUTS IN %s ***", compare_path)
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info(
+            "**************************************************************************"
+            "****"
+        )
+        LOG.info("")
+        from glob import glob
+        from os.path import basename, exists, isdir, isfile, join
+
+        compare_basenames = [basename(yy) for yy in glob(compare_path + "/*")]
+        final_output_products = []
+
+        for output_product in output_products:
+            if isdir(output_product):
+                # Skip 'diff' output directory
+                continue
+
+            LOG.info(
+                "**********************************************************************"
+                "****"
+            )
+            LOG.info("*** COMPARE  %s ***", basename(output_product))
+            LOG.info(
+                "**********************************************************************"
+                "****"
+            )
+
+            if basename(output_product) in compare_basenames:
+                test_product_func = self.test_products
+                goodcomps, badcomps, compare_strings = test_product_func(
+                    output_product,
+                    join(compare_path, basename(output_product)),
+                    goodcomps,
+                    badcomps,
+                    compare_strings,
+                )
+            else:
+                missingcomps += [output_product]
+            final_output_products += [output_product]
+
+            # Make sure we leave things as we found them
+            if rezip is True:
+                self.gzip_product(output_product)
```

Comment:
Where does rezip get set ?

Perhaps all we want to do here is right at the beginning of this function if the extension is .gz, we gunzip then set rezip=True (so not global), then at the end if rezip is true, we rezip it.  So everything is completely internal to this "compare_outputs" function?

Also, I think what @jsolbrig was suggesting was that we could ALSO gunzip the test output (what is stored in the repo) when we actually do the comparison, then re-zip it - which would prevent the unzipped output sitting in the repo all the time.